agent: Fix terminal command truncation in confirmation dialogs

miguelcsx opened the pull request:
Closes #40322

When users run multi-line terminal commands, the confirmation dialog was only showing the first line with '- N more lines' truncated. This meant dangerous commands like 'rm -rf' or 'sudo' operations could be hidden from view, creating a security risk.

Now terminal commands always display in full, with scroll support for very long commands (20+ lines) to keep the confirmation buttons visible.

Other tools continue using truncation for better UI efficiency since they don't pose the same security concerns.

Release Notes:

- Agent: Fixed terminal command truncation in the agent panel to better expose long commands (e.g., potentially dangerous multi-line commands are now fully visible before execution)

miguelcsx committed:
agent: Fix terminal command truncation in confirmation dialogs (#40322)

When users run multi-line terminal commands, the confirmation dialog was
only showing the first line with '- N more lines' truncated. This meant
dangerous commands like 'rm -rf' or 'sudo' operations could be hidden
from view, creating a security risk.

Now terminal commands always display in full, with scroll support for
very long commands (20+ lines) to keep the confirmation buttons visible.

Other tools continue using truncation for better UI efficiency since
they don't pose the same security concerns.

Closes #40322

Release Notes:

- Fixed terminal command truncation in confirmation dialogs - dangerous
  multi-line commands are now fully visible before execution

danilo-leal commented:
Hey @miguelcsx, thanks for the PR! Quick question, though, how can I easily test this? I want to check the UI given that adding overflow scroll in the tool call container might cause double scroll, which is something we avoid having.

miguelcsx commented:
I guess is ready for review, I added a scroll handle to track the scrolling in the inner box. Fell free to play with before I make it ready to review.

Here is a preview: 

https://github.com/user-attachments/assets/64b95172-d74d-4adf-97c7-75fc3eaa9df2

danilo-leal committed:
Merge remote-tracking branch 'origin/main' into pr/40570

danilo-leal committed:
Refine design a bit

dinocosta committed:
Merge branch 'main' into refactor/message

dinocosta committed:
refactor: show one command per line

- Update the way that the `TerminalTool` displays its command, returning
  back to the version where newlines are actually displayed, so if a
  multi-line command is used, it will be displayed as multiple lines.
- Refactor the `AcpThreadView.render_terminal_command_preview` in order
  to show, at maximum, three lines of the command, with an expand button
  to show the full command if it is longer. The label in the expand
  button will inform the user of how many additional lines are hidden.

dinocosta committed:
refactor(agent): improve terminal command preview collapse behavior

- Replace `MarkdownElement` with manual line rendering for command text
- Only show expand/collapse button when command exceeds 3 lines
- Display "N more lines" label when collapsed instead of always showing button
- Remove `window` parameter from `render_terminal_command_preview`

dinocosta committed:
Merge branch 'main' into refactor/message

dinocosta committed:
test(agent_ui): add recent_projects and title_bar dev depdendencies

These changes ensure we're again able to just run `cargo test -p
agent_ui` in order to run the test suite for the `agent_ui` crate.

danilo-leal committed:
Merge remote-tracking branch 'origin/main' into pr/40570

danilo-leal committed:
Refine the UI more

danilo-leal committed:
Fix clippy

danilo-leal committed:
Add button to copy the command
